### PR TITLE
go-chromecast: fix build by using `buildGo122Module`

### PR DESCRIPTION
**Pull request by zi3m5f**
ZHF: #352882

Fails on `0.3.1` and `upstream/master` with:
> Running phase: checkPhase
> github.com/rogpeppe/go-internal/testscript
vendor/github.com/rogpeppe/go-internal/testscript/exe_go118.go:14:27: cannot use nopTestDeps{} (value of type nopTestDeps) as testing.testDeps value in argument to testing.MainStart: nopTestDeps does not implement testing.testDeps (missing method InitRuntimeCoverage)
FAIL    github.com/vishen/go-chromecast [build failed]

Probably caused by changes in go 1.23.

Other changes while at it:
- adopt (no hardware to test)
- add `versionCheckHook`
- add shell completions with `installShellFiles`
- nixfmt
- move to `pkgs/by-name`
- other improvements

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/go-chromecast/default.nix [hunk 1]
```diff
@@ -1,24 +1,44 @@
...
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+    "-X main.commit=${src.rev}"
+    "-X main.date=unknown"
+  ];
 
-  meta = with lib; {
+  doCheck = false; # fails to build
```
Comment (Aleksanaa) [inline pkgs/applications/video/go-chromecast/default.nix:29]:
```suggestion
  doCheck = false; # fails to build tests
```

**Review by Aleksanaa — APPROVED** [review]
Would be good to split other modifications to another commit

File: pkgs/applications/video/go-chromecast/default.nix [hunk 2]
```diff
@@ -1,24 +1,44 @@
...
 
   vendorHash = "sha256-cEUlCR/xtPJJSWplV1COwV6UfzSmVArF4V0pJRk+/Og=";
 
-  ldflags = [ "-s" "-w" "-X main.version=${version}" "-X main.commit=${src.rev}" "-X main.date=unknown" ];
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+    "-X main.commit=${src.rev}"
+    "-X main.date=unknown"
```
Comment (fabaff) [inline pkgs/applications/video/go-chromecast/default.nix:26]:
```suggestion
    "-X=main.version=${version}"
    "-X=main.commit=${src.rev}"
    "-X=main.date=unknown"
```

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/go/go-chromecast/package.nix [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      --bash <($out/bin/go-chromecast completion bash) \
+      --fish <($out/bin/go-chromecast completion fish) \
+      --zsh <($out/bin/go-chromecast completion zsh)
+  '';
+
+  meta = {
+    homepage = "https://github.com/vishen/go-chromecast";
+    description = "CLI for Google Chromecast, Home devices and Cast Groups";
+    license = lib.licenses.asl20;
+    maintainers = [ ];
```
Comment (Aleksanaa) [inline pkgs/by-name/go/go-chromecast/package.nix:55]: Would you like to maintain this package?

**Comment by fabaff** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353037`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>go-chromecast</li>
  </ul>
</details>

**Comment by Aleksanaa** [comment]
> I found out an alternative fix would be to just use go 1.22 with `buildGo122Module` instead. Would that be better? Sorry, no go expert ;)

A little better, I think. But we still hope that upstream will fix it eventually, because Go versions are updated very quickly and we will not provide support for deprecated versions.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**